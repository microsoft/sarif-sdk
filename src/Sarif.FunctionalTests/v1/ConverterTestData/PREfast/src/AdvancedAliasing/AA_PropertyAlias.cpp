#define WIN32_LEAN_AND_MEAN             // Exclude rarely-used stuff from Windows headers

#include <Windows.h>

// Also look for indirect aliases e.g. when a property (not value) is
// passed from one variable to another

void h1c(wchar_t * p)
{
    int i = 6;
    int j = sizeof(p);

    i = i+j;

    int k = i + wcslen(p);
}
