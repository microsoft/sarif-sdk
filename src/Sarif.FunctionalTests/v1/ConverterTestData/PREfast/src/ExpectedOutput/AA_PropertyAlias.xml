<?xml version='1.0' encoding='UTF-8'?>
<DEFECTS><DEFECT><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedaliasing\</FILEPATH><FILENAME>aa_propertyalias.cpp</FILENAME><LINE>15</LINE><COLUMN>8</COLUMN></SFA><DEFECTCODE>6305</DEFECTCODE><DESCRIPTION>Potential mismatch between sizeof and countof quantities. Use sizeof() to scale byte sizes.</DESCRIPTION><FUNCTION>h1c</FUNCTION><DECORATED>?h1c@@YAXPA_W@Z</DECORATED><FUNCLINE>8</FUNCLINE><PROBABILITY>1</PROBABILITY><RANK>2</RANK><CATEGORY><RULECATEGORY>mspft</RULECATEGORY></CATEGORY><PATH><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedaliasing\</FILEPATH><FILENAME>aa_propertyalias.cpp</FILENAME><LINE>10</LINE><COLUMN>8</COLUMN></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedaliasing\</FILEPATH><FILENAME>aa_propertyalias.cpp</FILENAME><LINE>11</LINE><COLUMN>8</COLUMN><KEYEVENT><ID>1</ID><KIND>declaration</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>'j' is initialized with a sizeof quantity (measured in bytes)</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedaliasing\</FILEPATH><FILENAME>aa_propertyalias.cpp</FILENAME><LINE>13</LINE><COLUMN>6</COLUMN><KEYEVENT><ID>2</ID><KIND>alias</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>(alias) 'i' receives a sizeof quantity (in bytes) from 'j'</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedaliasing\</FILEPATH><FILENAME>aa_propertyalias.cpp</FILENAME><LINE>15</LINE><COLUMN>8</COLUMN><KEYEVENT><ID>3</ID><KIND>usage</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>'i' is used in an expression that expects element counts, not bytes</MESSAGE></KEYEVENT></SFA></PATH></DEFECT></DEFECTS>